sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -537,10 +603,135 @@ def peek_messages(self, max_message_count=1, **kwargs):
         }
 
         self._populate_message_properties(message)
-
+        handler = functools.partial(mgmt_handlers.peek_op, receiver=self)
         return self._mgmt_request_response_with_retry(
             REQUEST_RESPONSE_PEEK_OPERATION,
             message,
-            mgmt_handlers.peek_op,
+            handler,
             timeout=timeout
         )
+
+    def complete_message(self, message):
+        """Complete the message.
+
+        This removes the message from the queue.
+
+        :param message: The received message to be completed.
+        :type message: ~azure.servicebus.ReceivedMessage
+        :rtype: None
+        :raises: ~azure.servicebus.exceptions.MessageAlreadySettled if the message has been settled.
+        :raises: ~azure.servicebus.exceptions.MessageLockExpired if message lock has already expired.
+        :raises: ~azure.servicebus.exceptions.MessageSettleFailed if message settle operation fails.
+        """
+        if not isinstance(message, ServiceBusReceivedMessage):
+            raise TypeError("Parameter 'message' must be of type ReceivedMessage")
+        self._check_message_alive(message, MESSAGE_COMPLETE)
+        self._settle_message_with_retry(message, MESSAGE_COMPLETE)
+        message._settled = True  # pylint: disable=protected-access
+
+    def abandon_message(self, message):
+        """Abandon the message.
+
+        This message will be returned to the queue and made available to be received again.
+
+        :param message: The received message to be abandoned.
+        :type message: ~azure.servicebus.ReceivedMessage
+        :rtype: None
+        :raises: ~azure.servicebus.exceptions.MessageAlreadySettled if the message has been settled.
+        :raises: ~azure.servicebus.exceptions.MessageLockExpired if message lock has already expired.
+        :raises: ~azure.servicebus.exceptions.MessageSettleFailed if message settle operation fails.
+        """
+        if not isinstance(message, ServiceBusReceivedMessage):
+            raise TypeError("Parameter 'message' must be of type ReceivedMessage")
```

Comment:
Error messages should be updated for ServiceBusReceivedMessage

Reply:
Also the parameter types in the docstrings

Reply:
thanks for pointing, added